packages/api-explorer/src/scenes/OAuthScene/OAuthScene.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect } from 'react'
+import { Looker40SDK } from '@looker/sdk/lib/browser'
+import { tryItSDK } from '@looker/try-it'
+import { Redirect } from 'react-router'
+
+interface OAuthSceneProps {
+  sdk?: Looker40SDK
+}
+
+export const OAuthScene: React.FC<OAuthSceneProps> = ({ sdk }) => {
+  if (!sdk) sdk = tryItSDK
+
+  useEffect(() => {
+    // TODO is async really this complicated? https://dev.to/alexandrudanpop/correctly-handling-async-await-in-react-components-part-2-4fl7
+    async function login() {
+      try {
+        await sdk!.authSession.login()
+      } catch (err) {
+        console.error(err)
+      }
+    }
+
+    login().then(() => console.log('OAuth login completed'))
+  })
+
+  // TODO display OAuth login completed with a click to redirect, and redirect automatically after 3 seconds or so?
+  // TODO Or just show "OAuth login completed" as a disappearing banner on the redirected page?
+  return <Redirect to="/" />
```

Comment:
We might want to redirect to `/:specKey` here otherwise the redirect will result in another redirect because of:
https://github.com/looker-open-source/sdk-codegen/blob/35e28c22e09471ca2a0efd24a4e14b4e2222282f/packages/api-explorer/src/routes/AppRouter.tsx#L54-L55
unless react-router does some optimization under the hood.

I know this is still a WIP but you'll probably also need a local state here to cause a rerender upon logging in. Perhaps something along the lines of:

```Typescript
export const OAuthScene: React.FC<OAuthSceneProps> = ({ sdk, specKey }) => {
   const [isAuthenticated, setIsAuthenticated] = useState(false)
   const [loading, setLoading] = useState(true)
    
   ...
 
   return (
   <>
     {loading && <Text>Establishing OAuth session...</Text>
     {!loading && !isAuthenticated && 
       <Text>
         Failed to establish OAuth session. Ensure your app is registered.
       </Text>
     // whichever scenario we end up going with from the two described in your TODO:
     {isAuthenticated && <Redirect to=`/${specKey} />} 
   </>
   )
}
```

Reply:
Thanks for the tips and sample code!